pom.xml

```diff
@@ -67,28 +67,66 @@ under the License.
   </distributionManagement>
 
   <properties>
-    <javaVersion>8</javaVersion>
-    <mavenVersion>3.2.5</mavenVersion>
+    <maven.compiler.target>11</maven.compiler.target>
+    <mavenVersion>4.0.0-alpha-6-SNAPSHOT</mavenVersion>
     <project.build.outputTimestamp>2022-06-18T21:03:59Z</project.build.outputTimestamp>
   </properties>
 
   <dependencies>
     <dependency>
       <groupId>org.apache.maven</groupId>
-      <artifactId>maven-plugin-api</artifactId>
+      <artifactId>maven-api-core</artifactId>
       <version>${mavenVersion}</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
-      <artifactId>maven-core</artifactId>
+      <artifactId>maven-toolchain-builder</artifactId>
       <version>${mavenVersion}</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.plugin-tools</groupId>
       <artifactId>maven-plugin-annotations</artifactId>
+      <version>3.8.1</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.sisu</groupId>
+      <artifactId>org.eclipse.sisu.plexus</artifactId>
+      <version>0.9.0.M2</version>
+    </dependency>
   </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-plugin-plugin</artifactId>
+        <version>3.8.1</version>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-invoker-plugin</artifactId>
+        <version>3.5.0</version>
+      </plugin>
+      <plugin>
```

Comment:
I don't think this is related to spotless, but rather the switch to the v4 api.
This is actually not really required, though the `ToolchainDiscoverer` service will only be available in maven 4, so imho, it makes sense.
I've just added a bunch of commits, so that makes things a bit harder to follow, but if you kinda ignore the first commit (which is  more or less https://github.com/apache/maven-toolchains-plugin/pull/13), the patch is smaller.

Reply:
To make things easier, I've rebased this PR on top of the v3 api.